version: '3.8'

services:

  redis:
    image: redis:7.0.11-alpine

  django:
    container_name: django
    build:
      context: ./django_celery
    command: python manage.py runserver 0.0.0.0:8000
    volumes:
      - ./django_celery:/usr/src/app
    ports:
      - "8001:8000"
    environment:
      - DEBUG=1
      - SECRET_KEY=django-insecure-a^yf81h#2a++3n$zg!hb)w#*o!z2a&e@ck+1r1sm02o*%*^v7-
      - ALLOWED_HOSTS=localhost,127.0.0.1
    depends_on:
      - redis

  celery:
    container_name: celery1
    build:
      context: ./django_celery
    command: celery --app=django_celery worker -l INFO
    volumes:
      - ./django_celery:/usr/src/app
    environment:
      - DEBUG=1
      - SECRET_KEY="django-insecure-a^yf81h#2a++3n$zg!hb)w#*o!z2a&e@ck+1r1sm02o*%*^v7-"
      - ALLOWED_HOSTS=localhost,127.0.0.1
    depends_on:
      - redis
      - django

  celery2:
    container_name: celery2
    build:
      context: ./celeryworker
      dockerfile: Dockerfile
    command: celery --app celerytask worker -l INFO
    volumes:
      - ./celeryworker:/usr/src/app
    depends_on:
      - redis
      - django
